tests/network/vmi_istio.go

```diff
@@ -0,0 +1,313 @@
+			}
+			vmi = newVMIWithIstioSidecar(ports)
+			service = newService()
+
+			By("Create NetworkAttachmentDefinition")
+			nad = &k8snetworkplumbingwgv1.NetworkAttachmentDefinition{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "istio-cni",
+				},
+				Spec: k8snetworkplumbingwgv1.NetworkAttachmentDefinitionSpec{},
+			}
+			nad, err = virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(tests.NamespaceTestDefault).Create(context.TODO(), nad, metav1.CreateOptions{})
+			ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+
+			By("Creating VMI and associated Service")
+			_, err = virtClient.CoreV1().Services(tests.NamespaceTestDefault).Create(context.Background(), service, metav1.CreateOptions{})
+			ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+			vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+			ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+
+			By("Waiting for VMI to be ready")
+			tests.WaitUntilVMIReady(vmi, console.LoginToCirros)
+		})
+		Describe("Inbound traffic", func() {
+			var job *batchv1.Job
+			table.DescribeTable("request to VMI", func(port int) {
+				job, err = createJobCheckingVMIReachability(vmi, port)
+
+				By("Waiting for the job to succeed")
+				Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).To(Succeed())
+			},
+				table.Entry("should reach VMI HTTP server on service declared port", svcDeclaredTestPort),
+				table.Entry("should reach VMI HTTP server on service undeclared port", svcUndeclaredTestPort),
+			)
+			Context("With PeerAuthentication enforcing mTLS", func() {
+				peerAuthenticationRes := schema.GroupVersionResource{Group: "security.istio.io", Version: "v1beta1", Resource: "peerauthentications"}
+				var peerAuthentication *unstructured.Unstructured
+				BeforeEach(func() {
+					peerAuthentication = &unstructured.Unstructured{
+						Object: map[string]interface{}{
+							"apiVersion": "security.istio.io/v1beta1",
+							"kind":       "PeerAuthentication",
+							"metadata": map[string]interface{}{
+								"name": "strict-pa",
+							},
+							"spec": map[string]interface{}{
+								"mtls": map[string]interface{}{
+									"mode": "STRICT",
+								},
+							},
+						},
+					}
+					_, err = virtClient.DynamicClient().Resource(peerAuthenticationRes).Namespace(tests.NamespaceTestDefault).Create(context.Background(), peerAuthentication, metav1.CreateOptions{})
+					ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
+				})
+				table.DescribeTable("request to VMI", func(port int) {
+					job, err = createJobCheckingVMIReachability(vmi, port)
+
+					By("Waiting for the job to fail")
+					Expect(tests.WaitForJobToSucceed(job, 480*time.Second)).NotTo(Succeed())
+
+					err = virtClient.DynamicClient().Resource(peerAuthenticationRes).Namespace(tests.NamespaceTestDefault).Delete(context.Background(), peerAuthentication.GetName(), metav1.DeleteOptions{})
+					ExpectWithOffset(1, err).ShouldNot(HaveOccurred())
```

Comment:
I think you should move this teardown to an `AfterEach` block.

Reply:
I removed the teardown from here completely and instead I clean the istio resources in BeforeTestCleanup.
As per https://github.com/kubevirt/kubevirt/pull/5200#discussion_r617235464
we should avoid using AfterEach because if error occurres in AfterEach, it doesn't get collected

Reply:
OK, I had missed the fact that you're removing it in https://github.com/kubevirt/kubevirt/pull/5200/files?file-filters%5B%5D=.go#diff-847cd06c1bf6ef597ea96894f8b32ecb4d074ad1bc785b39871e11fb6ab71b3dR1774. 

But anyway, I don't think this argument is that strong, since you're not collecting the logs of this type of entity (you need to update the k8s reporter for that to happen: https://github.com/kubevirt/kubevirt/blob/master/tests/reporter/kubernetes.go#L91).

Anyway, not trying to change your mind, it's good as it is (probably better :) ); just saying that since you're not dumping this info, you **could** have used an `AfterEach` .